spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/MultiRabbitIntegrationTests.java

```diff
@@ -0,0 +1,332 @@
+		rabbitTemplate.convertAndSend("messageDefaultBroker");
+		Mockito.verify(MultiConfig.DEFAULT_CONNECTION_FACTORY).createConnection();
+		Mockito.verify(MultiConfig.DEFAULT_CONNECTION).createChannel(false);
+
+		Mockito.verify(MultiConfig.CONNECTION_FACTORY_BROKER_B, Mockito.never()).createConnection();
+		Mockito.verify(MultiConfig.CONNECTION_BROKER_B, Mockito.never()).createChannel(false);
+		SimpleResourceHolder.bind(MultiConfig.ROUTING_CONNECTION_FACTORY, "brokerB");
+		rabbitTemplate.convertAndSend("messageToBrokerB");
+		SimpleResourceHolder.unbind(MultiConfig.ROUTING_CONNECTION_FACTORY);
+		Mockito.verify(MultiConfig.CONNECTION_FACTORY_BROKER_B).createConnection();
+		Mockito.verify(MultiConfig.CONNECTION_BROKER_B).createChannel(false);
+
+		Mockito.verify(MultiConfig.CONNECTION_FACTORY_BROKER_C, Mockito.never()).createConnection();
+		Mockito.verify(MultiConfig.CONNECTION_BROKER_C, Mockito.never()).createChannel(false);
+		SimpleResourceHolder.bind(MultiConfig.ROUTING_CONNECTION_FACTORY, "brokerC");
+		rabbitTemplate.convertAndSend("messageToBrokerC");
+		SimpleResourceHolder.unbind(MultiConfig.ROUTING_CONNECTION_FACTORY);
+		Mockito.verify(MultiConfig.CONNECTION_FACTORY_BROKER_C).createConnection();
+		Mockito.verify(MultiConfig.CONNECTION_BROKER_C).createChannel(false);
+
+		context.close(); // Close and stop the listeners
+	}
+
+	@Component
+	static class AutoBindingListenerTestBeans {
+
+		@RabbitListener(bindings = @QueueBinding(
+				exchange = @Exchange("testExchange"),
+				value = @Queue("testQueue"),
+				key = "testKey"))
+		public void handleIt(String body) {
+		}
+
+		@RabbitListener(containerFactory = "brokerB", bindings = @QueueBinding(
+				exchange = @Exchange("testExchangeB"),
+				value = @Queue("testQueueB"),
+				key = "testKeyB"))
+		public void handleItB(String body) {
+		}
+
+		@RabbitListener(containerFactory = "brokerC", bindings = @QueueBinding(
+				exchange = @Exchange("testExchangeC"),
+				value = @Queue("testQueueC"),
+				key = "testKeyC"))
+		public void handleItC(String body) {
+		}
+	}
+
+	@Component
+	static class SimpleMessageListenerTestBean {
+
+		@RabbitListener(queues = "testQueue")
+		public void handleIt(String body) {
+		}
+
+		@RabbitListener(queues = "testQueueB", containerFactory = "brokerB")
+		public void handleItB(String body) {
+		}
+
+		@RabbitListener(queues = "testQueueC", containerFactory = "brokerC")
+		public void handleItC(String body) {
+		}
+	}
+
+	@Configuration
+	@PropertySource("classpath:/org/springframework/amqp/rabbit/annotation/queue-annotation.properties")
+	static class MultiConfig {
+
+		static final SimpleRoutingConnectionFactory ROUTING_CONNECTION_FACTORY = new SimpleRoutingConnectionFactory();
+		static final ConnectionFactory DEFAULT_CONNECTION_FACTORY = Mockito.mock(ConnectionFactory.class);
+		static final ConnectionFactory CONNECTION_FACTORY_BROKER_B = Mockito.spy(ConnectionFactory.class);
```

Comment:
Typo? There is nothing to `spy` on the interface.

Also: doesn't look like an `integration` test if you used mock.
Maybe better to call it `MockMultiRabbitTests`?